plugin/cql/src/main/java/org/opencds/cqf/ruler/cql/KnowledgeArtifactProcessor.java

```diff
@@ -52,6 +60,26 @@ public class KnowledgeArtifactProcessor {
 	public static final String CPG_INFERENCEEXPRESSION = "http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-inferenceExpression";
 	public static final String CPG_ASSERTIONEXPRESSION = "http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-assertionExpression";
 	public static final String CPG_FEATUREEXPRESSION = "http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-featureExpression";
+	private BundleEntryComponent createEntry(IBaseResource theResource) {
+		return new Bundle.BundleEntryComponent()
+				.setResource((Resource) theResource)
+				.setRequest(createRequest(theResource));
+	}
+
+	private BundleEntryRequestComponent createRequest(IBaseResource theResource) {
+		Bundle.BundleEntryRequestComponent request = new Bundle.BundleEntryRequestComponent();
+		if (theResource.getIdElement().hasValue() && !theResource.getIdElement().getValue().contains("urn:uuid")) {
+			request
+					.setMethod(Bundle.HTTPVerb.PUT)
+					.setUrl(theResource.getIdElement().getValue());
+		} else {
+			request
+					.setMethod(Bundle.HTTPVerb.POST)
+					.setUrl(theResource.fhirType());
+		}
```

Comment:
These are duplicates of helpers I added in the $approve operation, I'm not sure where to put them exactly to make it easy to reuse across the application